logdetective/server/server.py

```diff
@@ -147,40 +160,40 @@ async def submit_to_llm_endpoint(
     headers:
     stream:
     """
+    LOG.debug("async request %s headers=%s data=%s", url, headers, data)
     try:
-        # Expects llama-cpp server to run on LLM_CPP_SERVER_ADDRESS:LLM_CPP_SERVER_PORT
-        response = requests.post(
+        response = await http.post(
             url,
             headers=headers,
-            data=json.dumps(data),
+            # llama-server doesn't accept the encoding that data param does
```

Comment:
This comment is confusing. What is the encoding that the data parameter uses, why doesn't llama-server handle it, and what are we doing to work around it?

Reply:
I actually don't know what exactly has happened but the encoding that aiohttp performed using `data=` always failed with the server not being able to json-decode the request (unexpected char at position 1), even when the header was explicitly set to json; no idea what problem these two rascals had

once I changed it to `json=data`, all worked smoothly

now that I read this, let me update the comment